### PR TITLE
Fix ManagedNodeGroup taints being wrongly set in userdata

### PR DESCRIPTION
**Pull request by flostadler**
The taints for the `ManagedNodeGroup` component were being wrongly calculated when using custom userdata.
That was the case because the EKS service uses different capitalization for the taint effect enum than the kubernetes API(e.g. `NO_SCHEDULE` vs `NoSchedule`).
When building the custom userdata we need to map the EKS style enums to kubernetes style enums, otherwise it doesn't work.

Fixing this also revealed that taint values being absent aren't correctly handled either. The change fixes that as well.

While this was found during the v3 beta testing ([see](https://github.com/pulumi/pulumi-eks/issues/1425#issuecomment-2408954568)), it's a bug on v2 as well. The bug fix is targeting v3 anyways because the node group and user data logic was heavily rewritten during the development of v3. Back porting that to v2 would be too big of an effort given that v3 is right around the corner.

### CODE REVIEW

**Review by rquitales — APPROVED** [review]

File: examples/cluster-py/__main__.py [hunk 1]
```diff
@@ -84,11 +84,15 @@
...
+coredns_version = aws.eks.get_addon_version(addon_name="coredns",
+    kubernetes_version=cluster3["eks_cluster"]["version"],
+    most_recent=True)
+
 coredns = eks.Addon(
     f"{project_name}-cluster3-coredns",
     cluster=cluster3,
     addon_name="coredns",
-    addon_version="v1.11.1-eksbuild.9",
+    addon_version=coredns_version.version,
```
Comment (t0yv0) [inline examples/cluster-py/__main__.py:95]: Unrelated fix looks like.

Comment (t0yv0) [reply examples/cluster-py/__main__.py:95]: But great! I was fixing this again as port of ops, your solution is more robust.

**Review by t0yv0 — APPROVED** [review]
